tests/test-sampling.cpp

```diff
@@ -259,13 +259,17 @@ int main(void) {
     test_typical({0.97f, 0.01f, 0.01f, 0.01f}, {0.97f}, 0.5f);
     test_typical({0.4f, 0.2f, 0.2f, 0.2f}, {0.2f, 0.2f, 0.2f}, 0.5f);
 
-    test_repetition_penalties({0.2f, 0.2f, 0.2f, 0.2f, 0.2f}, {0}, {0.25f, 0.25f, 0.25f, 0.25f, 0},   50.0f, 0.0f, 0.0f);
```

Comment:
I do not understand how these tests are intended to work, even before my changes. The only prior token here is `0`, so I would expect the resulting probability vector to be `{0, 0.25f, 0.25f, 0.25f, 0.25f}`, that is, the probability at index `0` to be penalized. Please help me understand what is going on here so I can make sure the code actually works correctly.